evennia/scripts/taskhandler.py

```diff
@@ -0,0 +1,191 @@
+
+        """
+        persistent = kwargs.get("persistent", False)
+        if persistent:
+            del kwargs["persistent"]
+            now = datetime.now()
+            delta = timedelta(seconds=timedelay)
+
+            # Choose a free task_id
+            safe_args = []
+            safe_kwargs = {}
+            used_ids = self.tasks.keys()
+            task_id = 1
+            while task_id in used_ids:
+                task_id += 1
+
+            # Check that args and kwargs contain picklable information
+            for arg in args:
+                try:
+                    dbserialize(arg)
+                except (TypeError, AttributeError):
+                    logger.log_err("The positional argument {} cannot be " \
+                            "pickled and will not be present in the arguments " \
+                            "fed to the callback {}".format(arg, callback))
+                else:
+                    safe_args.append(arg)
+
+            for key, value in kwargs.items():
+                try:
+                    dbserialize(value)
+                except (TypeError, AttributeError):
+                    logger.log_err("The {} keyword argument {} cannot be " \
+                            "pickled and will not be present in the arguments " \
+                            "fed to the callback {}".format(key, value, callback))
+                else:
+                    safe_kwargs[key] = value
+
+            self.tasks[task_id] = (now + delta, callback, safe_args, safe_kwargs)
+            self.save()
+            callback = self.do_task
+            args = [task_id]
+            kwargs = {}
+
+        return task.deferLater(reactor, timedelay, callback, *args, **kwargs)
+
+    def remove(self, task_id):
+        """Remove a persistent task without executing it.
+
+        Args:
+            task_id (int): an existing task ID.
+
+        Note:
+            A non-persistent task doesn't have a task_id, it is not stored
+            in the TaskHandler.
+
+        """
+        del self.tasks[task_id]
+        if task_id in self.to_save:
+            del self.to_save[task_id]
+
+        self.save()
+
+    def do_task(self, task_id):
+        """Execute the task (call its callback).
+
+        Args:
+            task_id (int): a valid task ID.
+
+        Note:
+            This will also remove it from the list of current tasks.
+
+        """
+        date, callback, args, kwargs = self.tasks.pop(task_id)
+        if task_id in self.to_save:
+            del self.to_save[task_id]
+
+        self.save()
+        callback(*args, **kwargs)
+
+    def create_delays(self):
+        """Create the delayed tasks for the persistent tasks.
+
+        Note:
+            This method should be automatically called when Evennia starts.
+
+        """
+        now = datetime.now()
+        for task_id, (date, callbac, args, kwargs) in self.tasks.items():
+            seconds = (date - now).total_seconds()
```

Comment:
Minor quibble: The standard way to make sure a value is bigger or equal to zero is to do `seconds = max(0, (data - now).total_seconds())`. You can then remove the `if` statement.